core/src/main/java/org/mule/retry/policies/AbstractPolicyTemplate.java

```diff
@@ -100,13 +100,14 @@ public RetryContext execute(RetryCallback callback, WorkManager workManager) thr
             }
             while (status.isOk());
 
+            context.setFailed(cause);
```

Comment:
In FTP Reconnection Strategy Scenario, we need to know if an InterruptedException was thrown and the FTPClient wasn't able to be created. As the exception is not propagated by the PolicyTemplate, al least it should to return this in the context